### PR TITLE
Add more progress bar styles and configurations.

### PR DESCRIPTION
**Pull request by ZeroCool940711**
This PR adds support for configuring more styles and behaviors of both the linear and circular progress bars, I basically went and made all the options from Quasar configurable using simple parameters since there aren't that many and all this was already possible using the builder pattern. 

I also made it so we can now set a `max` value in the linear progress bar and this would allow us to also use an integer or a string for the `size`, this is really important since before we were limited to using only a float from `0.0-1.0` which reduced the usability of the linear progress bar, it was possible to override this but it was hard to do consistently and introduced unncessary extra steps and duplicated code in the user side.

### CODE REVIEW

File: nicegui/elements/progress.py [hunk 1]
```diff
@@ -1,48 +1,124 @@
...
 class LinearProgress(ValueElement, TextColorElement):
-    VALUE_PROP = 'value'
-
-    def __init__(self,
-                 value: float = 0.0, *,
-                 size: Optional[str] = None,
-                 show_value: bool = True,
-                 color: Optional[str] = 'primary',
-                 ) -> None:
+    VALUE_PROP = "value"
```
Comment (falkoschindler) [inline nicegui/elements/progress.py:9]: Please use single quotes.

File: nicegui/elements/progress.py [hunk 2]
```diff
@@ -1,48 +1,124 @@
...
+
+    def __init__(
+        self,
+        value: float = 0.0,
+        *,
+        min: float = 0.0,  # pylint: disable=redefined-builtin
+        max: Optional[Union[str, int]] = 1.0,  # pylint: disable=redefined-builtin
```
Comment (falkoschindler) [inline nicegui/elements/progress.py:16]: Why `str`?
Why `int` and not `float`? The default value `1.0` violates the type annotation.
Why `Optional`? The value `None` isn't properly handled below.

File: nicegui/elements/progress.py [hunk 3]
```diff
@@ -1,48 +1,124 @@
...
+        size: Optional[Union[str, int]] = 20,
```
Comment (falkoschindler) [inline nicegui/elements/progress.py:17]: Why not allowing `float` sizes?

File: nicegui/elements/progress.py [hunk 4]
```diff
@@ -1,48 +1,124 @@
...
+        :param query: Whether the progress bar is in query mode (default: False).
+        :param instant_feedback: Whether to provide instant feedback for the progress bar (default: False).
+        :param stripe: Whether to show a stripe pattern on the progress bar (default: False).
         """
-        super().__init__(tag='q-linear-progress', value=value, on_value_change=None, text_color=color)
-        self._props['size'] = size if size is not None else '20px' if show_value else '4px'
+        if max > 1.0:
+            value = float(value / max)
+
+        super().__init__(
```
Comment (falkoschindler) [inline nicegui/elements/progress.py:58]: Please don't reformat lines that are unrelated to this PR.
There are more such format changes below.

File: nicegui/elements/progress.py [hunk 5]
```diff
@@ -1,48 +1,124 @@
...
+        :param rounded: Whether to use rounded corners for the progress bar (default: False).
+        :param animation_speed: The animation speed of the progress bar (default: 2100).
+        :param indeterminate: Whether the progress bar is indeterminate (default: False).
+        :param query: Whether the progress bar is in query mode (default: False).
+        :param instant_feedback: Whether to provide instant feedback for the progress bar (default: False).
+        :param stripe: Whether to show a stripe pattern on the progress bar (default: False).
         """
-        super().__init__(tag='q-linear-progress', value=value, on_value_change=None, text_color=color)
-        self._props['size'] = size if size is not None else '20px' if show_value else '4px'
+        if max > 1.0:
```
Comment (falkoschindler) [inline nicegui/elements/progress.py:55]: Why `if max > 1.0`? This logic should hold for other `max` as well.
But I don't like dividing the `value` here. When the user sets the `value` argument to something like 42 and reads the `value` property later, it's value will be something different, which is rather surprising.

File: nicegui/elements/progress.py [hunk 6]
```diff
@@ -1,48 +1,124 @@
...
+            tag="q-linear-progress",
+            value=value,
+            on_value_change=None,
+            text_color=color,
+        )
+        self._props["min"] = min
+        self._props["max"] = max
+
+        if isinstance(size, int):
+            size = str(size) + "px"
```
Comment (falkoschindler) [inline nicegui/elements/progress.py:68]: I'm not sure if I like the implicit `px` unit. Why not simply take `str` sizes like `ui.icon` does?

**Review by falkoschindler — CHANGES_REQUESTED** [review]
Thanks for the PR, @ZeroCool940711.
Please try to avoid re-formatting existing code. This complicates code reviews and pollutes the Git history.
Besides that, there are several points that need to be addressed. See my comments in the code.
In general I'm not sure if we really need all those parameters. I'll think about it once again. But if we introduce something like `indeterminant` or `query`, we should also add properties to read and write these flags later. See notification.py for an example of such properties.

File: nicegui/elements/progress.py (hunk 5 above)
Comment (ZeroCool940711) [reply nicegui/elements/progress.py:55]: The reason for the `max > 1.0` condition is that there seem to be a bug with some numbers in some specific places, for some reason even if using a float some numbers would be converted to integer without decimal places and the opposite would happen as well, so, I added that condition to make sure we get a consistent behavior when using integers higher than `1`.
About the issue when reading the value back, I thought about it but wasn't sure what to do with it so we could get back the value again as an integer without breaking something else, so, I was waiting for your input on what to do to fix that.

File: nicegui/elements/progress.py (hunk 6 above)
Comment (ZeroCool940711) [reply nicegui/elements/progress.py:68]: There are two ways I thought we could use the `size` arg, one using it as a simple string like on `ui.icon` and some other places but we might also want to use it as an integer for fast prototyping, its easier to use a number than to figure out what string to use for the right size, so, the `px` unit is to make sure that when we use an integer we use pixels and not some other unit but this can be changes. We could also have an argument for the default `unit` to use to make it even more customizable.

File: nicegui/elements/progress.py (hunk 2 above)
Comment (ZeroCool940711) [reply nicegui/elements/progress.py:16]: That's a typo, I meant to use `float` and `int`.

Comment (ZeroCool940711) [reply nicegui/elements/progress.py]: Oh, then I will remove the `Optional` from it and make it `Union[float, int]`, would that be okay?

Comment (ZeroCool940711) [reply nicegui/elements/progress.py]: TIL. I didn't know that. I will change it to be just `float` then, will see what other values need changing based on this.

Comment (ZeroCool940711) [reply nicegui/elements/progress.py]: Done
